src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,107 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setLogin("ALice");
+        user.setPassword("456lfkdgj");
+        user.setAge(22);
+    }
+
+    @Test
+    void emptyUser_NotOk() {
+        Assertions.assertThrows(RuntimeException.class, () -> registrationService.register(null));
+    }
+
+    @Test
+    void nullLogin_NotOk() {
+        user.setLogin(null);
+        Assertions.assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void emptyLogin_NotOk() {
+        user.setLogin("");
+        Assertions.assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void existingUser_NotOk() {
+        storageDao.get(user.getLogin());
+        storageDao.add(user);
+        User user2 = new User();
+        user2.setLogin(user.getLogin());
+        user2.setPassword("slkgnUR12");
+        user2.setAge(20);
+        Assertions.assertThrows(RuntimeException.class, () -> registrationService.register(user2));
+    }
```

Comment:
create two users: firstUser and secondUser (using numbers in variable names is a bad practice)
register them by calling `register()`, not `storageDao.add()`
the second time you call `register()` you should expect error to be thrown